### DISCUSSION TITLE
Guidance around tasks with side effects like deployments

### DISCUSSION THREAD

**Discussion by ianduvall** [vercel/turborepo#7609] [Help] [answered]
### Summary

The [When Not To Cache](https://turbo.build/repo/docs/core-concepts/caching/to-cache-or-not-to-cache#when-not-to-cache) section of the docs has a note:

""cache": false does not mean "always run!" It means "if this task is to be executed it will not be restored from cache." Do not rely on "cache": false to trigger side effects like deployments."

I understand why you would not want to cache tasks with side effects but I'm not sure how I *should* declare tasks with side effects if I shouldn't use `"cache": false`. What is the recommended approach here?

### Example

I would like to be able to run `npx turbo-ignore server --task=deploy`. How should I configure the `deploy` tasks in my `turbo.json` if I shouldn't use `"cache": false`?

**Comment by anthonyshew** [comment, marked as answer]
I think the wording might be confusing. What it means by "always run!" is referring to "Will it run?" instead of "Will it run in full, missing cache, when it executes?"

Assuming a `turbo.json` of:
```
{
  "$schema": "https://turbo.build/schema.json",
  "pipeline": {
    "build": {
      "dependsOn": ["^build"],
      "outputs": ["dist/**"]
    },
    "deploy": {
      "dependsOn": ["build"],
      "cache": false
    }
  }
}

```

If the pipeline you write is `turbo deploy`, then `deploy` will execute, in full, no matter what. But if you write `turbo build`, the `deploy` task won't run.

We've had some folks in the past be under the impression that the `turbo build` command would fire off the `deploy` task for the package but I think we've over-indexed on that explanation here and perhaps made it more confusing. I'll make sure we're more clear on this in the new version of the docs I'm writing.

Does that help?

**Reply by ianduvall** [reply]
Thanks for the clarification
